include/silo/common/range.h

```diff
@@ -0,0 +1,85 @@
+#pragma once
+
+#include <cassert>
+#include <cstddef>
+#include <string>
+
+#include <fmt/format.h>
+
+/// A range of values that support `-` yielding `size_t`, `+ (size_t)`
+/// yielding `T`, `<=`, `++` and `==` (todo: find and use the concepts
+/// for these).
```

Comment:
Reminder: do it :)

Reply:
Sure, but not now? I'll pick that up after having studied C++ concepts more. I set add another kind of marker or reminder of course.

Reply:
Then let's at least add an issue number to the TODO so that the backlog reminds us of doing this.

Reply:
(I'll get back to it anyway, and using concepts is something that concerns the whole code base not just this. Add an issue when actually wanting to work on it.)